contracts/staking/fuse/FuseStaking.sol

```diff
@@ -243,54 +243,64 @@ contract FuseStaking is
 		debtToStakers /= PRECISION;
 		debtToDAO /= PRECISION;
 
+		uint256 totalAmountOfFuseForFuseAcceptingFaucets = _getAmountOfFuseForAllFaucets();
+
 		uint256 earnings = _balance() - debtToStakers - debtToDAO;
 		uint256 stakersPartInFuse = (earnings * (RATIO_BASE - globalGivebackRatio)) /
 			RATIO_BASE + debtToStakers;
-		uint256 daoPartInFuse = earnings - stakersPartInFuse + debtToDAO;
 
-		uint256 totalAmountOfFuseForFuseAcceptingFaucets = _getAmountOfFuseForAllFaucets();
-		uint256 totalFuseToSwap = stakersPartInFuse
-			+ daoPartInFuse - totalAmountOfFuseForFuseAcceptingFaucets;
+		uint256 keeperPartInFuse = (earnings - stakersPartInFuse)
+			- ((earnings - stakersPartInFuse) * (RATIO_BASE - keeperRatio)) / RATIO_BASE;
 
-		uint256[] memory buyResult = _safeBuyGD(
-			totalFuseToSwap,
-			keccak256("totalFuseToSwap")
-		);
+		uint256 daoPartInFuse = (earnings - stakersPartInFuse)
+				- keeperPartInFuse
+				+ debtToDAO
+				- totalAmountOfFuseForFuseAcceptingFaucets;
 
-		debtToStakers = buyResult[2] * PRECISION * stakersPartInFuse / totalFuseToSwap;
-		debtToDAO = buyResult[2] * PRECISION * daoPartInFuse / totalFuseToSwap;
+		uint256 totalFuseToSwap = stakersPartInFuse + daoPartInFuse + keeperPartInFuse;
 
-		uint256 stakersPartInGoodDollar = buyResult[1] * (RATIO_BASE - globalGivebackRatio)
-			/ RATIO_BASE;
+		uint256[] memory buyResult = _buyGD(totalFuseToSwap);
+
+		// avoiding the stack too deep error
+		buyResult[2] = totalFuseToSwap - buyResult[1];
 
-		uint256 daoPartInGoodDollar = buyResult[1] - stakersPartInGoodDollar;
+		debtToStakers = buyResult[2] * PRECISION * stakersPartInFuse / (totalFuseToSwap - keeperPartInFuse);
+		debtToDAO = buyResult[2] * PRECISION * daoPartInFuse / (totalFuseToSwap - keeperPartInFuse);
 
-		uint256 keeperPartInGoodDollar = daoPartInGoodDollar
-			- (daoPartInGoodDollar * (RATIO_BASE - keeperRatio)) / RATIO_BASE;
+		uint256 stakersPartInGoodDollar = buyResult[1] * PRECISION * stakersPartInFuse / totalFuseToSwap;
+		uint256 daoPartInGoodDollar = buyResult[1] * PRECISION * daoPartInFuse / totalFuseToSwap;
+		uint256 keeperPartInGoodDollar = buyResult[1] - (stakersPartInGoodDollar + daoPartInGoodDollar);
```

Comment:
this needs rethinking. because keeper needs to always get his FULL part, no debt.
so i'm going to simplify this. we wont convert keeper part to G$ we'll just send him his fuse. so modify code here and previously to reflect that

Reply:
done